### ISSUE TITLE
Question for git pull and how to insert - update sql when updating files ?

### ISSUE THREAD

**Issue by spirogg** [MacWarrior/clipbucket-v5#136]
@MacWarrior  
**you wrote** 
```
To simplify your usage and update process, you should really consider using git to copy sources :
git checkout https://github.com/MacWarrior/clipbucket-v5.git will create an updated copy of repo
git pull will update an existing copy of repo
```

- So I've managed to get git to work on my cPanel account, I copied all files and folders from upload to my public_html directory created mysql database etc and installed CB 
- all worked well configured ffmpeg MP4Box ffprobe mediainfo etc uploaded video and it works uploaded photo and it worked.

**QUESTION:** **_How to Update when you make changes?_**

1) Should I just upload the files that changed to proper folders one by one ? 
or
2) Copy the upload folder to public_html directory and delete install_cb folder?

- I always see you write to others  >>> DON'T FORGET TO UPDATE SQL? 

1) how would I do that I am not sure how to do that?

2) can you help me so I can learn that process

- **One Last Question** 

- if I upload all folders and files to current install I see offline mode

1) how do we turn back to online mode?

thanks for all your help
Spiro 😁

**Comment by MacWarrior** [comment]
Hi, 

1) With git, use your terminal and get to your CB directory (With cd command)
Once in it, use git : `git pull`
And then you're done, sources should have been updated to last revision

You can keep install_cb directory after install : as long as the file files/temp/install.me don't exist, this directory can't be used

2) For now, there is no functionnal upgrade system : this means CB is not able to self-update DB after initial installation
That's why when I push a new DB modification, already initialized CB have to manually apply the SQL modifications to be fully updated

3) To manually apply SQL you have to edit this SQL and replace {tbl_prefix} with the table prefix specified during installation and then apply the SQL to your DB (default prefix is cb_)
This can be done by multiple ways : with command line `mysql`, or with a DB software like phpMyAdmin, SQLyog, Navicat, etc....
From my side, to work on CB, I only use phpMyAdmin to access and edit my DB 

For example with commit c3104814f40b2d2685e65dfcbb3ed9214ba20c04 , you have to apply this : 
```ALTER TABLE `{tbl_prefix}contacts` MODIFY COLUMN `contact_group_id` INT(255) NOT NULL DEFAULT '0';```
After replacement : 
```ALTER TABLE `cb_contacts` MODIFY COLUMN `contact_group_id` INT(255) NOT NULL DEFAULT '0';```

4) Updating "correctly" sources should not enable an offline mode
To enable/disable offline mode, go to your admin area > General > Website configuration : Website closed

Best regards,
MacWarrior.

**Comment by spirogg** [comment]
ok im going to try this now, my other quick question is can you remove or rename and chmod this file - directory "https://github.com/MacWarrior/clipbucket-v5/commit/573b6da3d0443eadde9b5edec814d244720efd29" to a none execution ext.  can we  rename it? or in order to change password we can have a help my file that says please chanmge this file to .php in order to use it and then change back to .fff or whatever.

then when we do git pull this file will not be able to be executed if someone else tries to run it via our website..? 

- I also would like to ask, is there an easy way just to pull into public_html all files and folders in the (upload folder only) so then we do not have to copy files from another directory to public_html everytime we do a pull?

- we can just use pull and it will update our public_html folder with all the changes and then do the sql update in phpMyAdmin when you make a change ?

 is this possible somehow ? 


thanks so much

**Comment by spirogg** [comment]
> ALTER TABLE `{tbl_prefix}contacts` MODIFY COLUMN `contact_group_id` INT(255) NOT NULL DEFAULT '0';

ok so just to make sure I did this correctly

i went to PhpMyAdmin
- selected the databse for cb in the left side
- ![DB](https://user-images.githubusercontent.com/17736688/84229389-1239d480-aaaf-11ea-9f0f-2579994aad01.PNG)

- then up top I went to SQL
- then have a open filed where I can type in & copy and pasted this:

`ALTER TABLE `cb_contacts` MODIFY COLUMN `contact_group_id` INT(255) NOT NULL DEFAULT '0';`

- ![DB1](https://user-images.githubusercontent.com/17736688/84229431-27aefe80-aaaf-11ea-9ca8-8d80185d5c87.PNG)

- I then got a return that said below: 
```
(Check mark) MySQL returned an empty result set (i.e. zero rows). (Query took 0.0010 seconds.)
ALTER TABLE `cb_contacts` MODIFY COLUMN `contact_group_id` INT(255) NOT NULL DEFAULT '0'
```

**would this be correct the way  ?**

**Comment by MacWarrior** [comment]
Hi,

1) I forgot a little something in my explanations ; there is a little more to do than `git pull`
- `git reset --hard` This will remove all direct modification to sources to avoid conflict during pull
- `git pull` Now you know this one 😄
- `chmod 755 ./* -R` Setting a correct chmod to the sources (Be aware of this command, use it only IN the sources directory or edit the path in command)
- `chown www-data: ./* -R` Setting the correct owner to the sources (Like the one before, be aware of this command, use it only IN the sources directory or edit the path in command ; and also owner if needed)
I'll make an update script as example to make it easier

2) Which file do you reffer ? Can you specify the directory path ?

3) From my side I'm not working with any public_html directory ; we'll chat about it to make it clear for me ! 😄

4) You did everything right in phpMyAdmin and since you didn't get any error, you don't have to worry 😉

Best regards,
MacWarrior.

**Comment by spirogg** [comment]
2) Which file do you reffer ? Can you specify the directory path ?
can't remember what I asked here lol sorry

3) From my side I'm not working with any public_html directory ; we'll chat about it to make it clear for me ! 😄

-minstead of www-data/ directory when using cPanel all your main websites have to be in folder public_html/ 

-  this is where you upload all files and folders for website to work. :)

**Comment by MacWarrior** [comment]
Hi, 

I understood the purpose of your public_html directory, but I didn't get why you have to move directories or file on each update
As I see it, you should have something like this : 

/public_html/clipbucket/upload
/public_html/clipbucket/utils
...

And your Vhost has to use /upload directory, and not just /public_html
This way, you just have to go to /public_html/clipbucket and do the update process (List of commands I did in last message)

Best regards,
MacWarrior.

**Comment by spirogg** [comment]
hello sorry for late response
you wrote: 
/public_html/clipbucket/upload
/public_html/clipbucket/utils


so my main website is at /public_html

so what i was referring to is that i have to go to /public_html/clipbucket/upload
then copy all folders and files to /public_html

so my question was is there a way to get git to pull only the folders and files from /clipbucket/upload directly to my /public_html folder so it will just upload the folders and files that way ? since I do not use a clipbucket/upload folders for my main site 

my site is at https://mysite.com ( which is /public_html )
 not 
https://mysite.com/clipbucket/upload (not /public_html/clipbucket/upload )

**Comment by MacWarrior** [comment]
Hello @spirogg, 

Issue was related with SQL update and a new update system has been implemented on version 5.5.0 - revision 169 ; since then, SQL updates are applied by integrated systems

Best regards,
MacWarrior.
